frontend/user_context.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react'
+
+import {post, get} from './communication'
+
+/** The currently logged in user, if any */
+export interface User {
+  email: string | null
+  active: boolean
+}
+
+const null_user: User = {
+  email: null,
+  active: false
+}
+
+/** The currently logged in user, if any, and functionality to log in and log out */
+export interface UserContext {
+  user: User
```

Comment:
I'm a bit skeptical if you see if you are actually logged in on if the `email` is null or string`, why not have `user: User | undefined` here?